### DISCUSSION TITLE
+list-theme categorize light/dark themes separately.

### DISCUSSION THREAD

**Discussion by yam-liu** [ghostty-org/ghostty#4148] [Feature Requests, Ideas] [answered]
It hurts eyes when quickly switching one by one from dark and light backgrounds when picking up a theme.

**Comment by jcollie** [comment]
The question is how would this affect the startup time of `+list-themes`? You'd need to read every theme from disk at the start to categorize them. It _probably_ won't be too bad, but people can be sensitive to those things.

**Reply by mitchellh** [reply]
Related: https://github.com/ghostty-org/ghostty/pull/4026

**Comment by yam-liu** [comment]
Startup time is not the first thing to consider I think.

Easy to select and user friendly are.

Or we can pre cache the info about light or dark when new themes added.

Jeffrey C. Ollie ***@***.***>于2024年12月31日 周二13:04写道：

**Comment by jcollie** [comment]
See #4026

**Comment by rhodes-b** [comment, marked as answer]
https://github.com/ghostty-org/ghostty/pull/7248

this impliments a flag to choose between light and dark themes
